sympy/core/power.py

```diff
@@ -1677,16 +1677,19 @@ def _eval_nseries(self, x, n, logx, cdir=0):
             e = logcombine(e).cancel()
 
         if not (m.is_zero or e.is_number and e.is_real):
-            res = exp(e*log(b))._eval_nseries(x, n=n, logx=logx, cdir=cdir)
-            if res is exp(e*log(b)):
-                return self
-            return res
+            if self == self._eval_as_leading_term(x, logx=logx, cdir=cdir):
+                res = exp(e*log(b))._eval_nseries(x, n=n, logx=logx, cdir=cdir)
+                if res is exp(e*log(b)):
+                    return self
```

Comment:
It isn't safe to use `is` instead of `==` like this. This only works because of the cache.

Reply:
It looks like this is intended to catch the case where `exp._eval_nseries` would return `self`. But there may be other cases where the nseries is not well defined. I don't think that we should rely too much on `self` as a return value in problematic cases. It is more like a provisional arrangement.

Reply:
> But there may be other cases where the nseries is not well defined.

As in ... Could you maybe give any example ?

> I don't think that we should rely too much on self as a return value in problematic cases.

Hmm, we return `self`, when `res` is returned back in form `exp(e*log(b))` and `res` if this is not the case . The following test is being answered by the block
```
def test_issue_14384():
    x, a = symbols('x a')
    assert series(x**a, x) == x**a
    assert series(x**(-2*a), x) == x**(-2*a)
    assert series(exp(a*log(x)), x) == exp(a*log(x))
    assert series(x**I, x) == x**I
    assert series(x**(I + 1), x) == x**(1 + I)
    assert series(exp(I*log(x)), x) == exp(I*log(x))
```` 
Not sure how we could improve this . Do you know of some case which we will miss out but should be answered by this block currently .